deployment-scanner/src/main/java/org/jboss/as/server/deployment/scanner/FileSystemDeploymentService.java

```diff
@@ -152,6 +153,11 @@ class FileSystemDeploymentService implements DeploymentScanner, NotificationHand
     @SuppressWarnings("deprecation")
     private final DeploymentTransformer deploymentTransformer;
 
+    private ScanContext scanContext;
```

Comment:
Making this an instance variable is a bit dangerous for maintenance, so it needs a good usage comment. Something like

// Guarded by scanLock. 
// Reads or writes of this field can only be done in code executing between a call to {@link #acquireScanLock()} and {@link #releaseScanLock()}